### ISSUE TITLE
the ItemChangedType doesn't pass correct value to it's EditTemplate

### ISSUE THREAD

**Issue by cairabbit** [dotnetcore/BootstrapBlazor#4721]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I'm using Table to manage data with custom EditTemplate. My edit template component EditItemPanel has a field:

```
@typeparam TItem
@inject IStringLocalizer<EditItemPanel<TItem>> L

@code {
  [CascadingParameter]
  public ItemChangedType ChangedType { get; set; }
  [Parameter]
  [EditorRequired]
  public required TItem Model { get; set; }
  [Parameter]
  public string? NewTitle { get; set; }
  [Parameter]
  public string? EditTitle { get; set; }
  [Parameter]
  public RenderFragment<TItem>? NewItemTemplate { get; set; }
  [Parameter]
  [EditorRequired]
  public required RenderFragment<TItem> EditItemTemplate { get; set; }
}

<Panel>
  <Header>
    <div>
      @if (ChangedType == ItemChangedType.Add)
      {
        <h3>@(NewTitle ?? L.GetString("New Item"))</h3>
      }
      else
      {
        <h3>@(EditTitle ?? L.GetString("Edit Item"))</h3>
      }
    </div>
    <div class="flex-fill"></div>
    <div>
      <DialogCloseButton Icon="fa-solid fa-xmark" ButtonStyle="ButtonStyle.Round" Size="Size.Small" Text=""></DialogCloseButton>
    </div>
  </Header>
  <Body>
    <ValidateForm Model="@Model">
      @if (ChangedType == ItemChangedType.Add)
      {
        if (NewItemTemplate != null)
        {
          @NewItemTemplate(Model)
        }
        else
        {
          @EditItemTemplate(Model)
        }
      }
      else
      {
        @EditItemTemplate(Model)
      }
    </ValidateForm>
  </Body>
</Panel>
```

Then I use this compoent in table:

```
<Table TItem="AuthApp"
       IsPagination="true" PageItemsSource="@pageItemsSource" DataService="@dataService" CustomerSearchModel="filters"
       IsStriped="true" IsBordered="false" IsMultipleSelect="true"
       ShowToolbar="true" ShowSearch="true" ShowExtendButtons="true" ShowSkeleton="true" CollapsedTopSearch="true"
       ShowEmpty="true" SearchMode="SearchMode.Popup" SearchDialogSize="Size.Medium" EditMode="EditMode.Drawer"
       ShowAddButton="true" ShowExtendEditButtonCallback="@(o => true)" ShowExtendDeleteButtonCallback="@(o => true)">
  <TableColumns>
       ......
  </TableColumns>
  <CustomerSearchTemplate>
    .....
  </CustomerSearchTemplate>
  <EditTemplate>
    <EditItemPanel Model="@context" NewTitle="@L.GetString("New App")" EditTitle="@L.GetString("Edit App")">
      <EditItemTemplate Context="it">
        ......
      </EditItemTemplate>
    </EditItemPanel>
  </EditTemplate>
</Table>
```  

When clicking on the extension button "edit" on each row, the ChangedType is always Add value. 

### Expected Behavior

Clicking on the "edit" button on each row, it should pass ItemChangedType.Update to the EditTemplate's child component.

### Interactive render mode

Interactive Server (Interactive server-side rendering (interactive SSR) using Blazor Server)

### Steps To Reproduce

see above description

### .NET Version

9.0

**Comment by ArgoZhang** [comment]
@cairabbit could you submit a repro-project for this issue? thanks

**Comment by cairabbit** [comment]
Here is the repro-project: https://github.com/cairabbit/BlazorExample


https://github.com/user-attachments/assets/fb4e6f0a-8bfd-4e23-8ccb-b8ff344b54b5

**Comment by ArgoZhang** [comment]
@cairabbit fix in the version [9.0.2-beta04](https://www.nuget.org/packages/BootstrapBlazor/9.0.2-beta04)

**Comment by seasontian** [comment]
@ArgoZhang Below code In Table.razor forced to only pass ItemChangedType.Update in EditForm mode, popup mode still not work... please check, thank you!

```
if (!IsExcel && EditMode == EditMode.EditForm && ShowEditForm && SelectedRows.LastOrDefault() == item)
{
    @RenderEditForm((EditModel, ItemChangedType.Update))
}
```
